src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -514,12 +514,27 @@ protected function initializeContainer()
         $cache = new ConfigCache($this->getCacheDir().'/'.$class.'.php', $this->debug);
         $fresh = true;
         if (!$cache->isFresh()) {
-            $container = $this->buildContainer();
+            if ($this->debug) {
+                $collectedLogs = array();
+                $previousHandler = set_error_handler(function () use (&$collectedLogs, &$previousHandler) {
+                    $collectedLogs[] = $args = func_get_args();
+
+                    return $previousHandler ? call_user_func_array($previousHandler, $args) : false;
+                });
+            }
+
             try {
+                $container = null;
+                $container = $this->buildContainer();
                 $container->compile();
             } finally {
                 if ($this->debug) {
-                    file_put_contents($this->getCacheDir().'/'.$class.'Compiler.log', implode("\n", $container->getCompiler()->getLog()));
+                    restore_error_handler();
+                    file_put_contents($this->getCacheDir().'/'.$class.'BootLogs.log', serialize($this->cleanBotLogs($collectedLogs)));
```

Comment:
lets just add the `.log` extension? the BootLogs looks like duplicated info to me